test/libsolidity/SemanticTest.cpp

```diff
@@ -271,9 +277,37 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 		return TestResult::Failure;
 	}
 
+	if (m_gasCostFailure)
+	{
+		AnsiColorized(_stream, _formatted, {BOLD, CYAN})
+			<< _linePrefix << "Gas results missing or wrong, obtained result:" << endl;
+		for (auto const& test: m_tests)
+		{
+			ErrorReporter errorReporter;
+			_stream << test.format(errorReporter, _linePrefix, false, _formatted) << endl;
+			_stream << errorReporter.format(_linePrefix, _formatted);
+		}
+		return TestResult::Failure;
+	}
+
 	return TestResult::Success;
 }
 
+bool SemanticTest::checkGasCostExpectation(TestFunctionCall& _test, bool _compileViaYul) const
```

Comment:
`_test.call()` is `const` so I think you should be able to make the `_test` parameter const too.

Otherwise it should be called `io_test` according to our convention (which is ugly but does bring the attention to the fact that it's being modified).

```suggestion
bool SemanticTest::checkGasCostExpectation(TestFunctionCall const& _test, bool _compileViaYul) const
```

Reply:
There is a call to 1setGasCost1 in that function, that is why there is no `const`
`_test.setGasCost(setting, m_gasUsed);`

Reply:
Renamed to `io_test` to match convention